app/models/collection.js

```diff
@@ -45,11 +58,14 @@ class Collection {
     }
 
     save() {
-        this.bullets = this.bullets.map(bullet => bullet.id);  //beforeSave, converts bullet instances to ids
-        return db.rel.save('collection', this);
+        const bulletInstances = this.bullets;
+        this.serializeBullets();
```

Comment:
Again, I find this kind of messy. If you had two properties on the collection—one for the bullet ids and one for the bullet instances, could that work?